interface/src/lib.rs

```diff
@@ -340,11 +424,49 @@ impl<'a> MockHost<'a> {
     }
   }
 
-  pub fn deploy_code(&mut self, address: Address, code: &'a [u8]) {
-    self.programs.insert(address, code);
+  pub fn new_with_context(
+    static_context: HostStaticContext,
+    dynamic_context: HostDynamicContext,
+  ) -> Self {
+    MockHost {
+      dynamic_context: Some(dynamic_context),
+      static_context: Some(static_context),
+      ..MockHost::default()
+    }
+  }
+
+  pub fn spawn_program(
+    &mut self,
+    template: &Address,
+    blob: Vec<u8>,
+    principal: &Address,
+    nonce: u64,
+  ) -> Address {
+    let address = calculate_address(template, &blob, principal, nonce);
+    self.programs.insert(address, blob.clone());
+    self.spawn_result = Some(SpawnResult {
+      address,
+      blob,
+      template: *template,
+      principal: *principal,
+      nonce,
+    });
+    address
   }
 
-  fn transfer_balance(&mut self, from: &Address, to: &Address, value: u64) -> StatusCode {
+  pub fn get_spawn_result(&self) -> Option<&SpawnResult> {
+    self.spawn_result.as_ref()
+  }
```

Comment:
Because the spawn host call cannot currently return anything. None of the host calls have return values (https://github.com/athenavm/athena/blob/vmsdk/vm/hostfunctions/src/lib.rs). We probably want to change this: #81.